students/lauraannf/Lesson03/mailroom.py

```diff
@@ -0,0 +1,56 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Mon Sep 24 13:06:38 2018
+
+@author: Laura.Fiorentino
+"""
+DonorList = [
+        'Frank Reynolds', [10, 20, 50], 
+        'Dee Reynolds', [25, 100], 
+        'Dennis Reynolds', [10, 50], 
+        'Mac McDonald', [25, 35, 20], 
+        'Charlie Kelly', [0.25]
+        ]
+
+def namefunc(Name):
+    while Name == 'list' or Name == 'List':
+                print(DonorList[0::2])
+                Name = input("Type full donor name, or ''list'' for choices?>")
+    if Name not in DonorList:
+        DonorList.append(Name)
+        Donation = input("Donation Amount?>")
+        Donation = float(Donation)
+        DonorList.append([Donation])      
+    else:
+        indx =  DonorList.index(Name)
+        Donation = input("Donation Amount?>")
+        Donation = float(Donation)
+        DonorList[indx+1].append(Donation)
+    print('Email Creation:')
+    print('Dear {}, \n'
+          'Thank you so much for your generous donation of ${:.2f} \n'
+          'Sincerely, \n'
+          'Laura F'.format(Name,Donation))
+def reportfunc():
+    print('-------List of Donors-------')
+    print('{:<20}{:<20}{:<20}{:<20}'.format('Donor Name', 'Total Donated', '# of Donations', 'Average Donation'))
+    print('-----------------   '*4)
+    for it in range(0,len(DonorList),2):
+        print('{:<20}${:<20.2f}{:<20d}${:<20.2f}'.format(DonorList[it], sum(DonorList[it+1]), len(DonorList[it+1]),  sum(DonorList[it+1])/len(DonorList[it+1])))              
+    
+def main():
+    print('----------Mailroom------------')
+    Task = input("Choose an action: [1] Send a Thank You; [2] Create a Report; [3] Quit)>")
+    while Task not in ('1','2','3'):
+        Task = input("Choose an action: [1] Send a Thank You; [2] Create a Report; [3] Quit)>")
+    while Task !='3':  
+        if Task == '1':
+            Name = input("Type full donor name, or ''list'' for choices?>")
+            namefunc(Name)
+            Task = input("Choose an action: [1] Send a Thank You; [2] Create a Report; [3] Quit)>")
+        elif Task == '2':
+            reportfunc()
+            Task = input("Choose an action: [1] Send a Thank You; [2] Create a Report; [3] Quit)>")
```

Comment:
here is a modified entry point based on your code:
```
while True:
    task = input("choose an action: ...")
    if task == '1':
        send_thank_you()
    elif task == '2':
        create_report()
    elif task == '3':
        break
```
notice how much simpler the code is ^^^

here are the reasons:
* you have lines 43, 45, 50, and 53 all repeated - although you could use a variable for the string it is still an overkill; 
* line 48 should be moved inside of thank you functionality, in lesson4 it will be pretty much a requirement
* namefunc and reportfunc don't need "func" as part of it.. it is similar to naming "task_var" / "name_var"; it is known already that you are calling a function, therefore redundant.

Reply:
Thank you, I was having a difficult time with the "while" but I think it's starting to become more clear.

Reply:
👍 

what helps me is to write the code first just to make it work and then iterate on it - by that I mean, look over it again and ask yourself: "is there a duplication?" "can the logic be simplified". Almost always it is the case.